src/transformers/models/blip/processing_blip.py

```diff
@@ -0,0 +1,149 @@
+# coding=utf-8
+# Copyright 2022 The HuggingFace Inc. team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Processor class for Blip.
+"""
+
+from typing import List, Optional, Union
+
+from ...processing_utils import ProcessorMixin
+from ...tokenization_utils_base import BatchEncoding, PaddingStrategy, PreTokenizedInput, TextInput, TruncationStrategy
+from ...utils import TensorType
+
+
+class BlipProcessor(ProcessorMixin):
+    r"""
+    Constructs a BLIP processor which wraps a BERT tokenizer and BLIP image processor into a single processor.
+
+    [`BlipProcessor`] offers all the functionalities of [`BlipImageProcessor`] and [`BertTokenizerFast`]. See the
+    docstring of [`~BlipProcessor.__call__`] and [`~BlipProcessor.decode`] for more information.
+
+    Args:
+        feature_extractor (`BlipImageProcessor`):
+            An instance of [`BlipImageProcessor`]. The feature extractor is a required input.
+        tokenizer (`BertTokenizerFast`):
+            An instance of ['BertTokenizerFast`]. The tokenizer is a required input.
+    """
+    feature_extractor_class = "BlipImageProcessor"
+    tokenizer_class = ("BertTokenizer", "BertTokenizerFast")
+
+    def __init__(self, feature_extractor, tokenizer):
```

Comment:
wondering if we can make this `image_processor` straightaway instead of `feature_extractor`

Reply:
cc @amyeroberts

Reply:
I think this has to be done on `ProcessorMixin` level but not sure

Reply:
proposed a fix in [7ba9cb0](https://github.com/huggingface/transformers/pull/20716/commits/7ba9cb0f1b5455b5edfd5f0516ee1b84a106c4f2)

Reply:
Yes - we should have `image_processor` and `image_processor_class` directly on new processors. Old processors also have `image_processor` and `image_processor_class` with deprecation of the old FE references e.g. [for CLIP](https://github.com/huggingface/transformers/blob/main/src/transformers/models/clip/processing_clip.py#L38-L57)